core/src/main/java/org/opensearch/sql/planner/physical/UnnestOperator.java

```diff
@@ -67,6 +67,35 @@ public UnnestOperator(PhysicalPlan input, List<Map<String, ReferenceExpression>>
     );
   }
 
+  /**
+   * Constructor for UnnestOperator only when ".*" is included in the
+   * argument for the nested function
+   *
+   * @param input : PhysicalPlan input.
+   * @param fields : List of all fields and paths for nested fields.
+   * @param projectList : ProjectList to allow the operator to extract all fields
+   */
+  public UnnestOperator(
+      PhysicalPlan input,
+      List<Map<String, ReferenceExpression>> fields,
+      List projectList) {
+    this.input = input;
+    this.fields = (Set<String>) projectList.stream()
+        .map(e -> e.toString()).collect(Collectors.toSet());
+    this.groupedPathsAndFields = new HashMap<>();
+    List<String> paths = fields.stream().map(path -> path.get("path")
+        .getAttr()).collect(Collectors.toList());
+
+    for (String path : paths) {
+      this.groupedPathsAndFields.put(path, new ArrayList<>());
+      for (String field : this.fields) {
+        if (field.contains(path)) {
+          this.groupedPathsAndFields.get(path).add(field);
+        }
+      }
```

Comment:
```suggestion
      this.groupedPathsAndFields.put(path, 
          this.fields.stream().filter(f -> field.contains(path)).collect(Collectors.toList()));
```

Reply:
Changed in https://github.com/Bit-Quill/opensearch-project-sql/pull/256/commits/14f410c441df9436a0d4605c063923224bb32be8